codeSnippets/snippets/request-validation/src/main/kotlin/com/example/Application.kt

```diff
@@ -0,0 +1,66 @@
+package com.example
+
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import io.ktor.server.application.*
+import io.ktor.server.plugins.contentnegotiation.*
+import io.ktor.server.plugins.requestvalidation.*
+import io.ktor.server.plugins.statuspages.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import kotlinx.serialization.*
+
+@Serializable
+data class Customer(val id: Int, val firstName: String, val lastName: String)
+
+fun main(args: Array<String>): Unit = io.ktor.server.netty.EngineMain.main(args)
+
+fun Application.module() {
+    install(RequestValidation) {
+        validate<String> { bodyText ->
```

Comment:
These validations are artificial and in most cases won't be encountered in real-world applications.